### PR TITLE
Moving Juvix into Proper Components

### PR DESCRIPTION
**Pull request by mariari**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,6 +18,25 @@ build-z3:
 build:
 	stack build --copy-bins --fast -j $(shell nproc)
 
+build-cache:
```
Comment (cwgoes) [inline Makefile:21]: hmm can we still do this with a master `stack.yaml` file?

File: Makefile [hunk 2]
```diff
@@ -34,9 +53,25 @@ format:
 	find . -path ./.stack-work -prune -o -path ./archived -prune -o -type f -name "*.hs" -exec ormolu --mode inplace {} --ghc-opt -XTypeApplications --ghc-opt -XUnicodeSyntax --ghc-opt -XPatternSynonyms --ghc-opt -XTemplateHaskell \;
 
 org-gen:
-	org-generation app/ doc/Code/App.org test/ doc/Code/Test.org src/ doc/Code/Juvix.org bench/ doc/Code/Bench.org
+	org-generation app/ doc/Code/App.org test/ doc/Code/Test.org src/ doc/Code/Juvix.org bench/ doc/Code/Bench.org library/ doc/Code/Library.org
 
 test:
+	cd library/StandardLibrary; \
```
Comment (cwgoes) [inline Makefile:59]: ditto

also if we're going to repeat commands like this you can abstract it in a Makefile variable

File: library/Backends/ArithmeticCircuit/arithmetic-circuit.cabal [hunk 3]
```diff
@@ -0,0 +1,91 @@
+cabal-version: 1.12
```
Comment (cwgoes) [inline library/Backends/ArithmeticCircuit/arithmetic-circuit.cabal:1]: this file should not be committed, since we are using stack / hpack

Comment (cwgoes) [reply library/Backends/ArithmeticCircuit/arithmetic-circuit.cabal:1]: ditto for subsequent files

**Review by cwgoes — CHANGES_REQUESTED** [review]
A few comments, haven't yet tested locally.

File: Makefile (hunk 1 above)
Comment (mariari) [reply Makefile:21]: you can type stack build... however I think the master `stack.yaml` flag has some extra flags(?) it seems to ruin my cache at least for some reason.

File: library/Backends/ArithmeticCircuit/arithmetic-circuit.cabal (hunk 3 above)
Comment (mariari) [reply library/Backends/ArithmeticCircuit/arithmetic-circuit.cabal:1]: my bad, it got through, as I'm often adding files fast

File: stack.yaml [hunk 4]
```diff
@@ -98,8 +106,8 @@ extra-deps:
 
 allow-newer: true
 
-extra-lib-dirs:
-  - .stack-work/prefix/lib
+# extra-lib-dirs:
+#   - .stack-work/prefix/lib
```
Comment (cwgoes) [inline stack.yaml:110]: this was for Z3 but I guess we can figure it out later

**Review by cwgoes — CHANGES_REQUESTED** [review]
Just a few more nits and still need to abstract the Makefile `test-all` command

File: Makefile [hunk 5]
```diff
@@ -34,11 +34,30 @@ format:
...
+	stack test --fast --jobs=$(shell nproc) --test-arguments "--hide-successes --ansi-tricks false"
+	cd library/Translate; \
+	stack test --fast --jobs=$(shell nproc) --test-arguments "--hide-successes --ansi-tricks false"
+	cd library/Backends/LLVM; \
+	stack test --fast --jobs=$(shell nproc) --test-arguments "--hide-successes --ansi-tricks false"
+	cd library/Backends/Michelson; \
+	stack test --fast --jobs=$(shell nproc) --test-arguments "--hide-successes --ansi-tricks false"
+	cd library/Backends/ArithmeticCircuit; \
+	stack test --fast --jobs=$(shell nproc) --test-arguments "--hide-successes --ansi-tricks false"
+	stack test --fast --jobs=$(shell nproc) --test-arguments "--hide-successes --ansi-tricks false"
```
Comment (cwgoes) [inline Makefile:59]: this line is duplicated

Comment (mariari) [reply Makefile:59]: it is not!, it is to build the main, and not arith circuit!

Comment (cwgoes) [reply Makefile:59]: ah I see

File: Makefile (hunk 2 above)
Comment (cwgoes) [reply Makefile:43]: doing this in a separate PR

**Review by cwgoes — APPROVED** [review]
ACK

**Pull request merged**